cylc/flow/cfgspec/globalcfg.py

```diff
@@ -1559,6 +1567,7 @@ def load(self) -> None:
 
         self._set_default_editors()
         self._no_platform_group_name_overlap()
+        validate_platforms()
```

Comment:
It would be better to pass the configuration object through to ensure that we are validating the correct configuration (e.g. the validation is effectively disabled for `cached = False` ATM)